pkg/operator/generic_client.go

```diff
@@ -38,6 +39,10 @@ func (p *genericClient) GetOperatorState() (*operatorapiv1.OperatorSpec, *operat
 	return &instance.Spec.OperatorSpec, &instance.Status.OperatorStatus, instance.ResourceVersion, nil
 }
 
+func (p *genericClient) GetObjectMeta() (*metav1.ObjectMeta, error) {
```

Comment:
Had to add this function to be in compliance with being an OperatorClient

Reply:
Perhaps throwing an error here is bad?

Reply:
There is precedent in another repo (i can find it if you want me to) for returning the not implemented error here.

Reply:
Perhaps...but if we have a canonical instance of our object, and it has standard metadata, why not return the real thing?

Reply:
The struct for this has informers and a client, there is no metadata?

Reply:
Looking at the other methods that implement the interface, it seems that the Get/Update methods fetch the canonical operator config record from the lister:

```
resource, err := p.informers.Operator().V1().OpenShiftControllerManagers().Lister().Get("cluster")
...
```

The new method was added in https://github.com/openshift/library-go/pull/817. It appears this method is only used by some operators today, but not ours at present. IMO the implementation here is straightforward, so we should do it now vs. dealing with a bug down the road.

Reply:
updated in current push